src/growl.c

```diff
@@ -301,13 +303,17 @@ growl_tcp_register(
             goto leave;
         } else {
             int len = strlen(line);
-            /*fprintf(stderr, "%s\n", line);*/
-            if (strncmp(line, "GNTP/1.0 -ERROR", 15) == 0) {
-                if (strncmp(line + 15, " NONE", 5) != 0) {
-                    fprintf(stderr, "failed to register notification\n");
+            if (!strncmp(line, ERRSTR, strlen(ERRSTR))) {
+                fprintf(stderr, "Failed to register notification:\n%s\n", line);
+                free(line);
+                //Print full error-message
+                while((line = growl_tcp_read(sock)) != 0){
+                    fprintf(stderr, "%s\n", line);
+                    int br = (strlen(line) == 0);
```

Comment:
How can br be not true?
You already check for a null string, how can a string that is not null have a length of 0?